awswrangler/dynamodb/_utils.py

```diff
@@ -131,70 +166,136 @@ def execute_statement(
-        _execute_statement(kwargs=kwargs, boto3_session=boto3_session)
+        _execute_statement(kwargs=kwargs, dynamodb_client=dynamodb_client)
         return None
-    return _read_execute_statement(kwargs=kwargs, boto3_session=boto3_session)
-
-
-def _serialize_kwargs(kwargs: Dict[str, Any]) -> Dict[str, Any]:
-    """Serialize a DynamoDB input arguments dictionary.
+    return _read_execute_statement(kwargs=kwargs, dynamodb_client=dynamodb_client)
 
-    Parameters
-    ----------
-    kwargs : Dict[str, Any]
-        Dictionary to serialize.
 
-    Returns
-    -------
-    Dict[str, Any]
-        Serialized dictionary.
+def _validate_items(
+    items: Union[List[Dict[str, Any]], List[Mapping[str, Any]]], key_schema: List["KeySchemaElementTableTypeDef"]
+) -> None:
     """
-    names: Dict[str, Any] = {}
-    values: Dict[str, Any] = {}
-    serializer = TypeSerializer()
-
-    if "FilterExpression" in kwargs and not isinstance(kwargs["FilterExpression"], str):
-        builder = ConditionExpressionBuilder()
-        exp_string, names, values = builder.build_expression(kwargs["FilterExpression"], False)
-        kwargs["FilterExpression"] = exp_string
-
-    if "ExpressionAttributeNames" in kwargs:
-        kwargs["ExpressionAttributeNames"].update(names)
-    elif names:
-        kwargs["ExpressionAttributeNames"] = names
-
-    values = {k: serializer.serialize(v) for k, v in values.items()}
-    if "ExpressionAttributeValues" in kwargs:
-        kwargs["ExpressionAttributeValues"] = {
-            k: serializer.serialize(v) for k, v in kwargs["ExpressionAttributeValues"].items()
-        }
-        kwargs["ExpressionAttributeValues"].update(values)
-    elif values:
-        kwargs["ExpressionAttributeValues"] = values
-
-    return kwargs
-
-
-def _validate_items(items: Union[List[Dict[str, Any]], List[Mapping[str, Any]]], dynamodb_table: "Table") -> None:
-    """Validate if all items have the required keys for the Amazon DynamoDB table.
+    Validate if all items have the required keys for the Amazon DynamoDB table.
 
     Parameters
     ----------
-    items : Union[List[Dict[str, Any]], List[Mapping[str, Any]]]
+    items: Union[List[Dict[str, Any]], List[Mapping[str, Any]]]
         List which contains the items that will be validated.
-    dynamodb_table : boto3.resources.dynamodb.Table
-        Amazon DynamoDB Table object.
-
-    Returns
-    -------
-    None
-        None.
+    key_schema: List[KeySchemaElementTableTypeDef]
+        The primary key structure for the table.
+        Each element consists of the attribute name and it's type (HASH or RANGE).
     """
-    table_keys = [schema["AttributeName"] for schema in dynamodb_table.key_schema]
+    table_keys = [schema["AttributeName"] for schema in key_schema]
     if not all(key in item for item in items for key in table_keys):
         raise exceptions.InvalidArgumentValue("All items need to contain the required keys for the table.")
+
+
+class _TableBatchWriter:
```

Comment:
Can we keep using the [BatchWriter](https://github.com/boto/boto3/blob/fcc24f39cc0a923fa578587fcd1f781e820488a1/boto3/dynamodb/table.py#L63) from boto3? Is it going to be deprecated with the resources?